pkg/controller/storagecluster/reconcile.go

```diff
@@ -409,3 +434,34 @@ func newCephDaemonResources(custom map[string]corev1.ResourceRequirements) map[s
 
 	return resources
 }
+
+func (r *ReconcileStorageCluster) isActiveStorageCluster(instance *ocsv1.StorageCluster) (bool, error) {
+	storageClusterList := ocsv1.StorageClusterList{}
+	opts := &client.ListOptions{
+		Namespace: instance.Namespace,
+		Raw: &metav1.ListOptions{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       instance.Kind,
+				APIVersion: instance.APIVersion,
+			},
+		},
+	}
+
+	err := r.client.List(context.TODO(), opts, &storageClusterList)
+	if err != nil {
+		return false, fmt.Errorf("Error fetching StorageClusterList. %+v", err)
+	}
+
+	// There is no StorageCluster
+	if storageClusterList.Items == nil {
+		return true, nil
+	}
+	// There are many StorageClusters. Check if this is Active
+	for _, storageCluster := range storageClusterList.Items {
+		if storageCluster.Status.Phase != statusutil.PhaseIgnored &&
```

Comment:
There's a small chance that if two storageClusters were posted in a similar time frame that they'd both be marked as Phase=Ignore. 

The creationTimestamp and alphanumeric ordering as a tie breaker would prevent this. Realistically, i don't know if this is worth accounting for or not.

Reply:
Even if they're created at same time, Reconcile() will be called one at a time. The first one to be reconciled would be ignored and the second one would be marked active.

I don't think we need a tie breaker, unless we plan to Reconcile multiple StorageCluster at a time.

Reply:
However, i think we need to account for deletion immediately after creation so that we don't mark those CRs as active.

Reply:
> Even if they're created at same time, Reconcile() will be called one at a time. The first one to be reconciled would be ignored and the second one would be marked active.

maybe. This gets confusing because I'm unsure whether the "List" operation r.client performs is using internal cache of StorageClusters or actually making an api call to List all the current StorageClusters.

If the client is making an api call to List, then you're 100% correct.

If the client is using cache for the List, then it's possible the second entry will be marked as phase=Ignored as well.

Reply:
It most probably uses caching. Added tie-breaker just in case.

Reply:
Yes it uses the cache.

Reply:
> Reconcile() will be called one at a time

I would not depend on that. controller-runtime defaults to `MaxConcurrentReconciles` of 1, but that can be adjusted in-code. Of course we know right now not to do that given this code as-is, but in 6 or 12 months, will someone working on this code base know that its unsafe to enable concurrent reconciles? Better to keep it safe if possible.

Reply:
yeah good point. if we depend on 1 reconcile loop, we should explicitly set it to 1 then rather than depending on the default to remain as 1.

Furthermore if there's a way to unit test that we continue to use 1 that would be great as well. just something that catches if the value changes and forces whoever makes the change to update a unit test that clearly explains why "1" is used.

Reply:
Explicitly set `MaxConcurrentReconciles: 1`.
However, the logic doesn't care about it as each StorageCluster is independently validated and marked accordingly.